src/LegacyRecord.php

```diff
@@ -257,7 +259,7 @@ public function getCollection(): Collection
     /**
      * @throws BindingResolutionException
      */
-    public function bsonSerialize(): object|array
+    public function bsonSerialize(): array|stdClass|Document
```

Comment:
```suggestion
    public function bsonSerialize(): array
```
Analisando o `\Mongolid\Query\ModelMapper::map`, aparentemente ele só pode devolver um array

Reply:
Mudei aqui